sig-network charter (#271)
kubevirt/community: sig-network/charter.md

# SIG Network

## Scope

> SIG Network is owning everything related to KubevVirt networking.

Reviewer: KubevVirt typo

Author: done.

> The primary objective of the sig is to connect the Virtual Machine to the pod network/s,

Reviewer: maybe SIG ? (if so all over please)

Author: done.

> allowing migration and other advantages of virtualization in addition to maintaining services,

Reviewer: maybe add the word connectivity please ?
allowing "connectivity," migration....

Author: Connectivity is not a virtualization advantage. I tried to say we maintain k8s networking features alongside with virtualization ones.

Author: @ormergi Can I resolve this comment or is it a blocker from your side?

Reviewer: sgtm, you can close, thanks

port-forwarding, service meshes, and other Kubernetes capabilities.

### In scope
Some notable examples of features owned by SIG Network:
- Masquerade, Bridge and Passt binding plugins for the primary network.

> - Bridge, SRIOV and Macvtap bindings for secondary networks.

Reviewer: MacVTap (according wikipedia)

Author: done.

- VM networking lifecycle on creation, migration, hotplug/unplug, restart, stop etc.

#### Code, Binaries and Services

> - Networking related API on VM, VMI, Instance types, Kubevirt CR, etc.

Reviewer 2: ```suggestion
- Networking related APIs on: VM, VMI, Instance types, Kubevirt CR, etc.
```

Author: done.

- Core network binding infrastructure.
- Network binding plugin infrastructure.
- Examples of binding plugin CNIs and sidecars.
- Virtual Machine/Virtual Machine Instance networking status management.
- Domain networking configuration.

> - Unit and e2e tests

Reviewer: Please add dot, as the rest

Author: done.

### Out of scope
- Node networking, it is managed by [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate).
- Host to Pod connectivity, it is managed by external CNIs.

## Roles and Organization Management

This sig follows adheres to the Roles and Organization Management outlined in [OARP]

### Additional responsibilities of Chairs

There are some other projects managed by the SIG:

> - [Cluster Network Addon](https://github.com/kubevirt/cluster-network-addons-operator).

Reviewer: Please change the prefix to [Cluster Network Addon Operator]
because CNA is something else and can confuse

Author: done.

> - [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate).

Reviewer: Aint it mentioned above it is not part of sig-network ? confusing maybe ?
same for line below

Author: I"m listing here extra projects that are managed by the sig members but are not part of the sig.
But maybe that's not the intention of this paragraph.
@fabiand ?

Reviewer 3: If the SIG is owning it, then it should be named here.
IIUIC then nmstate is not owned by this SIG, thus it should not be here.

- [kubernetes-nmstate-policy](https://github.com/nmstate/nmpolicy).
- [KubeMacPool](https://github.com/k8snetworkplumbingwg/kubemacpool).

> - [KubeSecondaryDNS](https://github.com/kubevirt/kubesecondarydns)

Reviewer: maybe need to add please macvtap and dynamic-networks ?
what about OVS-CNI, bridge marker? (any other thing CNAO deploys and we forgot)

Author: added

# SIG Network

## Scope

SIG Network is owning everything related to KubeVirt networking.
The main objective of the SIG is to connect the Virtual Machine to the primary and secondary pod networks,
allowing migration and other advantages of virtualization in addition to maintaining services,
port-forwarding, service meshes, and other Kubernetes capabilities.

### In scope

> Some notable examples of features owned by SIG Network:

Author: done.

- Masquerade, Bridge and Passt binding plugins for the primary network.

> - Bridge, SRIOV and MacVTap bindings for secondary networks.

Author: We have it in `Code, Binaries and Services` section. Here I wanted to highlight some important features.

- VM networking lifecycle on: creation, migration, hotplug/unplug, restart, stop etc.

#### Code, Binaries and Services

> - Networking related APIs on: VM, VMI, Instance types, Kubevirt CR, etc.

Author: done.

- Core network binding infrastructure.
- Network binding plugin infrastructure.
- Examples of binding plugin CNIs and sidecars.
- Virtual Machine/Virtual Machine Instance networking status management.
- Domain networking configuration.

> - Unit and e2e tests.

Author: @EdDev since you lgtm-ed, resolving this comment.

- Developer and user documentation around networking.
- Help for onboarding new SIG Network members.
- Grooming network related bugs.

### Out of scope

> - Node networking, it is managed by [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate).

Author: done.

> - Host to Pod connectivity, it is managed by external CNIs.

Reviewer 4: This is partially correct. We do manage it as programmatic users of it (parse and write pod annotations, NAD configuration, etc).

Author: Changed it to `Host to Pod connectivity configuration`and added `Host to Pod connectivity declaration`to the `In Scope`.

## Roles and Organization Management

This sig follows adheres to the Roles and Organization Management outlined in [OARP]

> ### Additional responsibilities of Chairs

Reviewer 4: Did I miss where the "primary" repo was mentioned?

Author: I wasn't. Where do you think I should add it?

Reviewer 4: Maybe in `Code, Binaries and Services` ?

- Bridge, SRIOV and MacVTap bindings for secondary networks.
- VM networking lifecycle on: creation, migration, hotplug/unplug, restart, stop etc.

#### Code, Binaries and Services
- Networking related APIs on: VM, VMI, Instance types, Kubevirt CR, etc.
- Core network binding infrastructure.
- Network binding plugin infrastructure.
- Examples of binding plugin CNIs and sidecars.
- Virtual Machine/Virtual Machine Instance networking status management.
- Domain networking configuration.
- Unit and e2e tests.
- Developer and user documentation around networking.
- Help for onboarding new SIG Network members.
- Grooming network related bugs.

### Out of scope
- Node networking, it is managed by [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate).
- Host to Pod connectivity, it is managed by external CNIs.

## Roles and Organization Management

This sig follows adheres to the Roles and Organization Management outlined in [OARP]

### Additional responsibilities of Chairs
- Uphold the Kubrvirt Code of Conduct especially in terms of personal behavior and responsibility.
- Own sig-network CI jobs.
- There are some other projects written and managed by the SIG members:

>   - [Cluster Network Addon Operator](https://github.com/kubevirt/cluster-network-addons-operator).

Reviewer: s/Addon/Addons/ please

Author: done.

>   - [Cluster Network Addon Operator](https://github.com/kubevirt/cluster-network-addons-operator).

Author: done.

# SIG Network

## Scope

SIG Network is owning everything related to KubeVirt networking.

> The primary objective of the SIG is to connect the Virtual Machine to the pod network/s,

Reviewer 3: there is only one pod network, not?

Would it be fair to also include 2ndary networks in this charter?

Author: The pod may have primary and secondary networks. Changed to - `The main objective of the SIG is to connect the Virtual Machine to the primary and secondary pod networks,`.
I hope it is clearer.

allowing migration and other advantages of virtualization in addition to maintaining services,

> port-forwarding, service meshes, and other Kubernetes capabilities.

Reviewer 4: This is my suggestion to this section:

> SIG Network is responsible for all network capabilities in Kubevirt, with the high level goal of enabling Virtual Machine to connect to networks in the Kubernetes ecosystem.
> The SIG Network owns the API,  implementation and documentation related to all network functionalities.

Author: I liked better the previous option. Let's see what other reviewers think.

Reviewer: both are fine imho, but i am in favor of the current, it is bit more specific

Reviewer 4: IMO it is not a fit to make this specific, it is a charter which should not change when we add or remove features.

Said that, I will not insist further.

Reviewer 2: I like this suggestion, we can elaborate with more details at `Code, Binaries and Services`.

### In scope
Some notable examples of features owned by SIG Network:
- Masquerade, Bridge and Passt binding plugins for the primary network.
- Bridge, SRIOV and MacVTap bindings for secondary networks.
- VM networking lifecycle on: creation, migration, hotplug/unplug, restart, stop etc.

#### Code, Binaries and Services
- Networking related APIs on: VM, VMI, Instance types, Kubevirt CR, etc.
- Core network binding infrastructure.
- Network binding plugin infrastructure.
- Examples of binding plugin CNIs and sidecars.
- Virtual Machine/Virtual Machine Instance networking status management.
- Domain networking configuration.
- Unit and e2e tests.

> 

Reviewer 3: What about:
- Developer and user documentation around networking
- What about help for onboarding new sig network members?
- What about grooming network related bugs?

Author: Added. But I"m not sure how the last two fit the `Code, Binaries and Services`. Those are services?

- Developer and user documentation around networking.
- Help for onboarding new SIG Network members.
- Grooming network related bugs.

### Out of scope
- Node networking, it may be managed by [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate).
- Host to Pod connectivity configuration, it is managed by external CNIs.

## Roles and Organization Management

> This sig follows adheres to the Roles and Organization Management outlined in [OARP]

Reviewer 4: > follows adheres

I think you need to choose one verb.

- [Cluster Network Addon Operator](https://github.com/kubevirt/cluster-network-addons-operator).
- [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate).
- [kubernetes-nmstate-policy](https://github.com/nmstate/nmpolicy).

> - [KubeMacPool](https://github.com/k8snetworkplumbingwg/kubemacpool).

Reviewer: some are Camel notation some are all low in case it matters
some might be even `OVS CNI`

Author: done, all have the same format now.

- [KubeSecondaryDNS](https://github.com/kubevirt/kubesecondarydns).
- [ovs-cni](https://github.com/k8snetworkplumbingwg/ovs-cni).

> - [brdige-marker](https://github.com/kubevirt/bridge-marker).

Reviewer: bridge typo

Author: done.

> - [dynamic networks](https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller).

Reviewer: maybe better to call it please [multus dynamic networks controller] ?

Author: removed it since it is not under kubevirt.

# SIG Network

## Scope

> SIG Network is owning everything related to KubeVirt networking.

Reviewer 2: nit:
```suggestion
SIG Network owns KubeVirt networking topics.
```

Author: done.

> The main objective of the SIG is to connect the Virtual Machine to the primary and secondary pod networks,

Reviewer 2: nit:
```suggestion
The main objective of the SIG is to connect the guest to cluster services including: pod network, secondary networks,
```

Author: I prefer the original. I wanted to emphasize the connection to the primary and secondary networks  is the main goad. Using the word "including" gives the feeling it is yet another service.

Author: @ormergi Can I resolve this comment or is it a blocker from your side?

Reviewer 2: No blocker from my side, it a nit and I already approved this PR on previous review cycle :slightly_smiling_face: 
https://github.com/kubevirt/community/pull/271#pullrequestreview-1970215570

allowing migration and other advantages of virtualization in addition to maintaining services,

> port-forwarding, service meshes, and other Kubernetes capabilities.

Reviewer 2: nit:
```suggestion
port-forwarding, service-mesh, and other Kubernetes capabilities.
```

Author: done.

### In scope
Some notable examples of areas owned by SIG Network:
- Masquerade, Bridge and Passt binding plugins for the primary network.
- Bridge, SRIOV and MacVTap bindings for secondary networks.
- VM networking lifecycle on: creation, migration, hotplug/unplug, restart, stop etc.

#### Code, Binaries and Services
- [Main repo](https://github.com/kubevirt/kubevirt).
- Networking related APIs on: VM, VMI, Kubevirt CR, etc.
- Core network binding infrastructure.
- Network binding plugin infrastructure.
- Examples of binding plugin CNIs and sidecars.
- Virtual Machine/Virtual Machine Instance networking status management.
- Domain networking configuration.

> - Pod network definition.

Reviewer 2: This one is pretty vogue, does it mean connecting the guest to pod network?

Author: yes

Author: @ormergi Can I resolve this comment or do you want something to be changed here?

Reviewer 2: No blocker from my side, it a nit and I already approved this PR on previous review cycle :slightly_smiling_face: 
https://github.com/kubevirt/community/pull/271#pullrequestreview-1970215570

- Unit and e2e tests.
- Developer and user documentation around networking.
- Help for onboarding new SIG Network members.
- Grooming network related bugs.

### Out of scope
- Node networking, it may be managed by [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate).
- Host to Pod connectivity configuration, it is managed by external CNIs.

## Roles and Organization Management

> This sig adheres to the Roles and Organization Management outlined in [OARP]

Reviewer 2: Should OARP have URL link?

Author: The url is at the end, now I removed the double OARP tag it works.
It is part of the [sig-charter-template](https://github.com/kubevirt/community/blob/2d3384dafe8a1e1a6f3e780707c8283dbcb8791e/sig-TEMPLATE/sig-charter-template.md#L32-L31).

### Additional responsibilities of Chairs

> - Uphold the Kubrvirt Code of Conduct especially in terms of personal behavior and responsibility.

Reviewer 5: ```suggestion
- Uphold the KubeVirt Code of Conduct especially in terms of personal behavior and responsibility.
```

Author: done.

Author: done.

- Own sig-network CI jobs.
- There are some other projects written and managed by the SIG members:
  - [Cluster Network Addons Operator](https://github.com/kubevirt/cluster-network-addons-operator).
  - [Kube Secondary DNS](https://github.com/kubevirt/kubesecondarydns).
  - [Bridge Marker](https://github.com/kubevirt/bridge-marker).

> [OARP]: [OARP]: https://stumblingabout.com/tag/oarp/

Reviewer 2: Double OARP tags

Author: done.
